weave/integrations/openai_agents/openai_agents.py

```diff
@@ -0,0 +1,507 @@
+"""
+A Weave integration for OpenAI Agents.
+
+This module provides a TracingProcessor implementation that logs OpenAI
+Agent traces and spans to Weave.
+"""
+
+from __future__ import annotations
+
+from typing import Any, TypedDict
+
+from weave.integrations.patcher import NoOpPatcher, Patcher
+from weave.trace.autopatch import IntegrationSettings
+from weave.trace.context import call_context
+from weave.trace.context.weave_client_context import require_weave_client
+from weave.trace.weave_client import Call
+
+_openai_agents_patcher: OpenAIAgentsPatcher | None = None
+
+
+try:
+    from agents import tracing
+    from agents.tracing import TracingProcessor
+except ImportError:
+
+    class TracingProcessor: ...  # type: ignore[no-redef]
+
+
+def _call_type(span: tracing.Span[Any]) -> str:
+    """Determine the appropriate call type for a given OpenAI Agent span."""
+    return span.span_data.type or "task"
+
+
+def _call_name(span: tracing.Span[Any]) -> str:
+    """Determine the name for a given OpenAI Agent span."""
+    if name := getattr(span.span_data, "name", None):
+        return name
+    elif isinstance(span.span_data, tracing.GenerationSpanData):
+        return "Generation"
+    elif isinstance(span.span_data, tracing.ResponseSpanData):
+        return "Response"
+    elif isinstance(span.span_data, tracing.HandoffSpanData):
+        return "Handoff"
+    else:
+        return "Unknown"
+
+
+class WeaveDataDict(TypedDict):
+    inputs: dict[str, Any]
+    outputs: dict[str, Any]
+    metadata: dict[str, Any]
+    metrics: dict[str, Any]
+    error: dict[str, Any] | None
+
+
+class WeaveTracingProcessor(TracingProcessor):  # pyright: ignore[reportGeneralTypeIssues]
+    """
+    A TracingProcessor implementation that logs OpenAI Agent traces and spans to Weave.
+
+    This processor captures different types of spans from OpenAI Agents (agent execution,
+    function calls, LLM generations, etc.) and logs them to Weave as structured trace data.
+    Child spans are logged as separate calls but not redundantly included in the parent trace data.
+    """
+
+    def __init__(self) -> None:
+        self._trace_data: dict[str, dict[str, Any]] = {}
+        self._trace_calls: dict[str, call_context.Call] = {}
+        self._span_calls: dict[str, call_context.Call] = {}
+        self._ended_traces: set[str] = set()
+        self._span_parents: dict[str, str] = {}
+
+    def on_trace_start(self, trace: tracing.Trace) -> None:
+        """Called when a trace starts."""
+        # Set up basic trace data
+        self._trace_data[trace.trace_id] = {
+            "name": trace.name,
+            "type": "task",
+            "metrics": {},
+            "metadata": {},
+        }
+
+        # Create a call for this trace
+        wc = require_weave_client()
+        trace_call = wc.create_call(
+            op="openai_agent_trace",
+            inputs={"name": trace.name},
+            parent=call_context.get_current_call(),
```

Comment:
How is the call structure being resolved for a trace? They have group_id but only spans have a parent_id

Reply:
In this context, `trace_call` is basically a dummy call that acts like the "trace" container.

All call structure is otherwise modelled with child calls